src/java/voldemort/server/protocol/admin/AsyncOperationService.java

```diff
@@ -81,21 +81,28 @@ public synchronized void submitOperation(int requestId, AsyncOperation operation
      * @param requestId Id of the request
      * @return True if request is complete, false otherwise
      */
-    public synchronized boolean isComplete(int requestId) {
-        if(!operations.containsKey(requestId))
-            throw new VoldemortException("No operation with id " + requestId + " found");
+    public synchronized boolean isComplete(int requestId, boolean remove) {
+        //if requestId not found, return true
+        if (!operations.containsKey(requestId))
```

Comment:
I don't think it's a good idea to change this behavior. Instead, for the use case where you consider a non-existent requestId to be equivalent to a completed one, you should surround the call in a try catch and ignore the exception.